### PR TITLE
PanelEdit: Fixed timing and state related issues

### PR DESCRIPTION
**Pull request by torkelo**
Introduces a panels state that is now actually used more and a panelModelAndPluginReady redux action to mark a panel model + plugin ready (migrations run etc). 

This fixes timing issues in the panel edit when page reloading while in edit mode (panel + plugin had loaded but no migrations for this particular panel model)

### CODE REVIEW

File: public/app/features/dashboard/state/PanelModel.ts [hunk 1]
```diff
@@ -354,7 +354,12 @@ export class PanelModel {
   }
 
   getEditClone() {
-    const clone = new PanelModel(this.getSaveModel());
+    const sourceModel = this.getSaveModel();
+
+    // super unique id
+    sourceModel.id = 23763571993;
```
Comment (torkelo) [inline public/app/features/dashboard/state/PanelModel.ts]: this causes issues, need to restore id when changes are applied.

Comment (torkelo) [reply public/app/features/dashboard/state/PanelModel.ts]: and remove the panel state with this id from redux when edit mode is exited

File: public/app/features/dashboard/state/PanelModel.ts [hunk 2]
```diff
@@ -354,7 +354,12 @@ export class PanelModel {
   }
 
   getEditClone() {
-    const clone = new PanelModel(this.getSaveModel());
+    const sourceModel = this.getSaveModel();
+
+    // super unique id
```
Comment (dprokop) [inline public/app/features/dashboard/state/PanelModel.ts]:
```suggestion
    // Temporary id for the clone, restored later in redux action when changes are saved
```

**Review by dprokop — APPROVED** [review]
Looks solid! added some nit suggestion for comment in code

**Pull request merged**